### DISCUSSION TITLE
GitHub Student Dev Pack, and Global Campus access

### DISCUSSION THREAD

**Discussion by IMGROOT2** [For-0/vocabustudy#161] [General]
FYI: I’m ruhan lol, I followed @grimsteel like 5 mins ago

Hey Siddhant! I guess this is the place where I can contact you now lol. GitHub student benefits is so cool! You get JetBrains ULTIMATE (bro it’s like 50 dollars a month), GitHub pro, GITHUB COPILOT (bro it’s so cool! I installed it onto IntelliJ idea and it’s literally auto programming!), and so many free camps, events, custom domain names (very cool for vocabustudy), and so so much more. if you find any more cool things lmk because it’s really cool!

-ruhan

**Comment by grimsteel** [comment]
Exactly! For the past like 3 months I've been scrolling through that GitHub student benefits page wondering when I would ever be able to get access... Speaking of GitHub Copilot, IT LITERALLY KNOWS WHAT I WANT TO DO BEFORE I DO! I've heard so much about JetBrains/IntelliJ but I never really used it. It's a java ide right?

Also while the software is cool, you have to admit: Physical stuff is even cooler
![image](https://user-images.githubusercontent.com/115331356/197288639-6e0c5374-1980-4744-8eaa-259b68f3bf54.png) ![image](https://user-images.githubusercontent.com/115331356/197288862-f50c5731-fc7b-40ca-800d-6329dd97759d.png)
Actually nvm those discounts aren't that good. "50% off Arduino Due" - you can get it cheaper ~off the black market~ on a bunch of websites

### Some more things that seemed interesting: 
* **LingoHub** localization for free!! - 99 pounds per month value),
* **ConfigCat** feature flags - I can turn on/off features in Vocabustudy if something goes wrong - $99/month value
* **CryptoLens** software licensing/selling - if we want to make a product that people pay for (evil grin) then we can license it with this

About the domains: they seem to be on weird tlds (.me, .tech, .live)...

**Reply by IMGROOT2** [reply]
LOL yeah the benefits are insane. Although the endings to the domains aren’t the best, they’re still so much better than other possibilities.

IntelliJ IDEA is a Java ide, which is my go-to language, but jetbrains has a bunch of other IDEs, including some for c++, c, c#, ruby, js, Python (the python one is actually one of the best Python IDEs out there) and more. Idk about html/css, but I’m sure there’s probably one.

I literally joined a day ago and applied for benefits like right after LOL. But the value you get is at least $2000 a year if you use everything. JetBrains is 600 yearly alone I think (50 bucks a month)

**Reply by grimsteel** [reply]
Wow....

**Reply by grimsteel** [reply]
> Although the endings to the domains aren’t the best, they’re still so much better than other possibilities.
 
Yes...

**Reply by IMGROOT2** [reply]
![9E9031B3-3852-40A7-885E-CE7E0F88166A](https://user-images.githubusercontent.com/116324098/197293757-2c404255-54b4-4a0b-ba63-c012091e3a78.jpeg)

Oh my god it’s 800 wow

**Reply by grimsteel** [reply]
And that's just jetbrains...

**Reply by IMGROOT2** [reply]
Yeah! It’s really cool how they just give it to student developers for free

**Reply by grimsteel** [reply]
I feel like they have some ulterior motive...
I think the only reason we have G Suite for free is because google wants to brainwash us lol
We also apparently have office 365

**Reply by IMGROOT2** [reply]
yes, they do. Their motive is to "spread education to everyone" and "help poor children", but I think they just want to seem like a cool company that's not only just trying to make billions of dollars.

we have infinite storage on google drive on our school accounts, but I don't think we have office 365. we just have the normal online free web apps, and word 2019 installed on computer lab desktops (one time purchase, with product key, yk)

**Reply by IMGROOT2** [reply]
but... it benefits us! so, keep giving, please 😄

**Reply by grimsteel** [reply]
My sister installed Office 365 on her computer and it worked w/ her student account
https://tis.roundrockisd.org/office365/

> but... it benefits us! so, keep giving, please 😄

If it continues through college I will stay in college forever

**Reply by grimsteel** [reply]
> we have infinite storage

No there's actually a limit and it's not *that* high...
We literally only get 1024 petabytes

**Reply by IMGROOT2** [reply]
LOL! so maybe we have office 365 after all. nice!

my dad got a deal on the office 365 family subscription and it's really useful, you get like 1024 gb of storage and stuff

**Reply by IMGROOT2** [reply]
> No there's actually a limit and it's not _that_ high... We literally only get 1024 petabytes

![image](https://user-images.githubusercontent.com/116324098/197296958-a35b7f61-be35-434e-a824-5fc5a96af99f.png)
am I missing something

**Reply by grimsteel** [reply]
Wow!
From what I remember my dad got a deal on Office 2019 because he didn't want to have to sign in

**Reply by IMGROOT2** [reply]
![image](https://user-images.githubusercontent.com/116324098/197297048-ca43d712-0b1d-434c-83fb-319b95a5df7a.png)

**Reply by IMGROOT2** [reply]
Nice, that's really cool!

**Reply by grimsteel** [reply]
> am I missing something

Hmm I definitely remember seeing that limit in Google Drive but it changed to "Unlimited" now
Either way is there *that* much difference between 1024 pbs and unlimited?

**Reply by IMGROOT2** [reply]
I need to go get some stuff done lol. I might get onto github soon later. Bye!

**Reply by grimsteel** [reply]
Same I need to do HW.

**Reply by IMGROOT2** [reply]
> Hmm I definitely remember seeing that limit in Google Drive but it changed to "Unlimited" now Either way is there _that_ much difference between 1024 pbs and unlimited?

Yeah lol, not that much of a difference. Literally only google or Microsoft needs that much storage, and that's probably for renting it out!

**Comment by grimsteel** [comment]
Codespaces?
![image](https://user-images.githubusercontent.com/115331356/199636770-4e39fb7b-5a84-433a-aa38-fec175998581.png)
That just started appearing today but I can't seem to access Codespaces...

**Reply by grimsteel** [reply]
IDK if we need to turn it on in an org that a student owns (For Zero should be eligible. However, there's a separate offer for teachers where they can get orgs on the Pro plan for free) or if we should be able to create a codespace anywhere even in an org that doesn't have codespace access. Also, IDK what "Pro level access" means

If anyone's brave enough, we can report this on https://github.com/orgs/community/discussions/categories/codespaces or https://github.com/orgs/community/discussions/categories/github-education (probably the former)

**Reply by IMGROOT2** [reply]
@For-0 is an organization, but it is also an account, just like @grimsteel. AFAIK, @grimsteel is a Pro account, but @For-0 is a free account. So, @For-0 does not have codespaces, bc it is free.

**Reply by IMGROOT2** [reply]
such a missed opportunity ;( oh well

**Reply by grimsteel** [reply]
> @For-0 is an organization, but it is also an account, just like @grimsteel. AFAIK, @grimsteel is a Pro account, but @For-0 is a free account. So, @For-0 does not have codespaces, bc it is free.

I'm well aware - but Codespaces are not a user-level feature. From the docs, it seems that they are a feature that can exclusively be enabled on organizations. That's why I'm confused about the developer pack offer - "what does it mean by anywhere in your account"? I feel like we might still have a chance to get this!

**Reply by IMGROOT2** [reply]
i'll ask on the forums

**Reply by grimsteel** [reply]
I see you're braver than me :wink:

**Reply by IMGROOT2** [reply]
LOL https://github.com/orgs/community/discussions/38122

**Reply by grimsteel** [reply]
Thanks!

**Comment by grimsteel** [comment]
AAAAAHHHHHHHHHHHHHHHHHHHHH @IMGROOT2 LOOK:
![image](https://user-images.githubusercontent.com/115331356/200950786-ec1d1df7-30a9-44bb-ad2f-112a2e5308f6.png)

sorry I'm just super excited!!
There's a codespaces section in my personal profile, which seems to apply to all repos that I visit, even the ones I don't own!!

**Reply by IMGROOT2** [reply]
WAIT WHAT I WAS AT THE DENTIST LOL I JUST SAW THIS

WAIT SET IT UP ILL GET ON MY COMPUTER IN A MINUTE

**Reply by grimsteel** [reply]
You're going to have to use VS Code now 😈 (at least when on the codespace)
I'll get a "dev container" set up

**Reply by IMGROOT2** [reply]
DARN IT! 
![image](https://user-images.githubusercontent.com/116324098/200954531-0e9841e3-efda-4aa8-a6ed-71b68a3e925f.png)

i don't have access yet lol

**Reply by grimsteel** [reply]
I'm pretty sure the only part of the codespace that requires github.com is the login part, so if I can figure out how to work around that, we can do this at school too!

**Reply by grimsteel** [reply]
> i don't have access yet lol

I'm just special

**Reply by IMGROOT2** [reply]
> You're going to have to use VS Code now 😈 (at least when on the codespace)

Jokes on you, haha! 4 hours ago, jetbrains announced that you can use codespaces with jetbrains IDEs! I'm so excited

**Reply by IMGROOT2** [reply]
> I'm pretty sure the only part of the codespace that requires github.com is the login part, so if I can figure out how to work around that, we can do this at school too!

YESSS

**Reply by IMGROOT2** [reply]
> > i don't have access yet lol
> 
> I'm just special

LMAO

**Reply by IMGROOT2** [reply]
https://blog.jetbrains.com/blog/2022/11/09/remote-development-in-jetbrains-ides-now-available-to-github-codespaces-users/

**Reply by grimsteel** [reply]
> jetbrains IDEs

Get back to me when you figure out how to run those... things on chromebooks

**Reply by IMGROOT2** [reply]
> ...run those... things on chromebooks

WWIII: battle of the IDEs (or fake IDEs, a.k.a. VS Code 😎)

**Reply by grimsteel** [reply]
IDK why I am so against JetBrains...

I literally use Android Studio

**Reply by IMGROOT2** [reply]
HAHAHAH

**Reply by grimsteel** [reply]
> WWIII: battle of the IDEs (or fake IDEs, a.k.a. VS Code 😎)

That's WWIV. WWIII is Emacs vs Vim

**Reply by IMGROOT2** [reply]
you seem like someone who would code using vim for a living ngl

**Reply by grimsteel** [reply]
I used vim once, and tried doing `ctrl+c` to exit.

It took pity on me and gave me the essential knowledge
![image](https://user-images.githubusercontent.com/115331356/200956040-1d908cd2-c9f4-4385-955e-d35c3f1548a2.png)
The votes...

**Reply by IMGROOT2** [reply]
SIDDHANT DO WE GET PRO?!?

THAT'S LIKE CRAZY!

![image](https://user-images.githubusercontent.com/116324098/200956251-6df7f702-517f-4833-b284-eef1fddb1361.png)
![image](https://user-images.githubusercontent.com/116324098/200956257-fcb2013a-a43d-4909-8287-1ad72ce5677b.png)

also I attached the jetbrains editor thing haha

**Reply by grimsteel** [reply]
Well, we _do_ have Github Pro accts...

![real programmers](https://imgs.xkcd.com/comics/real_programmers.png)

**Reply by IMGROOT2** [reply]
LOL

**Reply by IMGROOT2** [reply]
have you set up codespaces?

**Reply by grimsteel** [reply]
No, I'm working on the dev container

**Reply by grimsteel** [reply]
![image](https://user-images.githubusercontent.com/115331356/201258630-03da6556-27eb-41f3-abd1-f636497e069e.png)
![image](https://user-images.githubusercontent.com/115331356/201258654-ac3ff8f3-d50b-4547-afb3-a055d26a587d.png)

**Comment by IMGROOT2** [comment]
@grimsteel I GOT IT!
![image](https://user-images.githubusercontent.com/116324098/201258036-5a1ba2c2-35d4-4236-8275-ab930877f6fb.png)

**Reply by grimsteel** [reply]
Nice!!

**Comment by ShreyanKhanna** [comment]
@IMGROOT2 @grimsteel I JUST GOT THE STUDENT DEV PACKKKK
but uh... Idk how to use it, cuz I have the email but I have no clue what it's supposed to do, or how I'm supposed to like claim everything.

**Comment by ShreyanKhanna** [comment]
Also @grimsteel I have some other ideas for applications to put under for0, IDK what im supposed to do with them, cuz I've heard that vocabustudy needs coders, with the amount of bugs and things that are left, but I sorta don't ant to forget about them, so,,,

**Reply by grimsteel** [reply]
You can help, if you want - just branch and PR

**Reply by ShreyanKhanna** [reply]
well, i do want to.. but like i said before its honestly rly confusing what to do cuz its hard to keep track of who is doing what at the moment right now.

**Reply by grimsteel** [reply]
Look for the issues without assignees

**Reply by ShreyanKhanna** [reply]
okay thanks

**Reply by ShreyanKhanna** [reply]
@grimsteel i was also wondering if you want to make it so that sets can only be given tags of one category, history, science, math, etc, so that we don't have US history sets in the math section, which is currently happening.

Alr I have also assigned myself to one of the issues I'll start working on asap.

**Reply by IMGROOT2** [reply]
alr cool, there's no need to constantly @grimsteel lol

**Reply by grimsteel** [reply]
I'm watching this repo so I get notified even if you don't @grimsteel (oops I did it again)
But as you know, all my notifications go in the trash

**Reply by IMGROOT2** [reply]
lol

**Reply by ShreyanKhanna** [reply]
Mb mb I’ll stop now
